Python/ceval.c

```diff
@@ -5319,8 +5385,9 @@ initialize_locals(PyThreadState *tstate, PyFrameConstructor *con,
     Py_ssize_t i;
     if (co->co_flags & CO_VARKEYWORDS) {
         kwdict = PyDict_New();
```

Comment:
Aren't there some possible leaks of kwdict at every `goto fail;`? The `fail:` label has no `Py_XDECREF(kwdict)`.

I'm not sure if that's why `(lambda **c: 0)(1)` leaks.

Reply:
Not technically, the `kwdict` goes to the end of the `localsplus` stack. The leaks are anything on the right of whatever we have copied so far when there is an error, as we increment the refcount of the whole stack for cleanup later.